packages/astro/src/@types/astro.ts

```diff
@@ -93,7 +93,7 @@ export interface BuildConfig {
  *
  * [Astro reference](https://docs.astro.build/reference/api-reference/#astro-global)
  */
-export interface AstroGlobal extends AstroGlobalPartial {
+export interface AstroGlobal<T = Record<string, number | string | any>> extends AstroGlobalPartial {
```

Comment:
Is there a reason to do `number | string | any`? Wouldn't `any` do the same thing?
Also, would be nice to have a more clear name (`Props`) and enforce the type via `extends`.

```suggestion
export interface AstroGlobal<Props extends Record<string, any> = Record<string, any>> extends AstroGlobalPartial {
```